#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.5574330011882082 -0.5880682516866498 -0.5860411082377365 2.131950850138055
  position -0.1157867440063937 1.4748239359039217 -0.0902889479356253
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation -0.02 0 -0.13
  name "rectangle arena(2)"
  floorAppearance Parquetry {
    type "dark strip"
  }
}
Robot {
  translation 0.347822 0.0574726 0.292599
  rotation 9.683719999199083e-06 -0.9999999999172924 -8.464099999299955e-06 -1.591115307179586
  children [
    HingeJoint {
      jointParameters HingeJointParameters {
        position 1811.623704578431
        axis -1 0 0
        anchor 0.045 -0.0176 0
      }
      device [
        RotationalMotor {
          name "motor roda esquerda"
        }
        PositionSensor {
          name "sensor roda esquerda"
        }
      ]
      endPoint Solid {
        translation 0.045004127923461694 -0.01759838189833128 2.1023247223233856e-07
        rotation 0.9999999512713679 -0.0002966836627049782 9.713941647571887e-05 4.741212474943255
        children [
          Transform {
            rotation 0 0 1 1.5707996938995747
            children [
              DEF shape_roda Shape {
                appearance PBRAppearance {
                  baseColor 0.8 0.15 0.3
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.02
                }
              }
            ]
          }
        ]
        name "roda direita"
        boundingObject Transform {
          rotation 0 0 1 1.5707996938995747
          children [
            USE shape_roda
          ]
        }
        physics DEF fisica_roda Physics {
          density -1
          mass 0.005
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 479.5924277792008
        axis -1 0 0
        anchor -0.045 -0.0176 0
      }
      device [
        RotationalMotor {
          name "motor roda direita"
        }
        PositionSensor {
          name "sensor roda direita"
        }
      ]
      endPoint Solid {
        translation -0.045 -0.0176 0
        rotation -1 0 0 2.070344433552246
        children [
          Transform {
            rotation 0 0 1 1.5707996938995747
            children [
              USE shape_roda
            ]
          }
        ]
        name "roda esquerda"
        boundingObject Transform {
          rotation 0 0 1 1.5707996938995747
          children [
            USE shape_roda
          ]
        }
        physics USE fisica_roda
      }
    }
    DEF rus_shape Shape {
      appearance PBRAppearance {
        baseColor 0.58 0.4 1
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.075 0.075 0.075
      }
    }
    DistanceSensor {
      translation 0 0.01 -0.038
      rotation 0 1 0 1.7016996938995748
      children [
        Transform {
          rotation 0 0 1 -1.5708003061004252
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.9 0.9 0
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ir0"
      lookupTable [
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
    }
    DistanceSensor {
      translation 0.0388758 0.01 0.00104763
      rotation 0 -1 0 0.130895
      children [
        Transform {
          rotation 0 0 1 -1.5708003061004252
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.9 0.9 0
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ir1"
      lookupTable [
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
    }
    DistanceSensor {
      translation -0.0404401 0.01 -0.00939408
      rotation 0 -1 0 0.130895
      children [
        Transform {
          rotation 0 0 1 -1.5708003061004252
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.9 0.9 0
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ir2"
      lookupTable [
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
    }
  ]
  name "rus"
  boundingObject DEF rus_shape Shape {
    appearance PBRAppearance {
      baseColor 0.58 0.4 1
      roughness 1
      metalness 0
    }
    geometry Box {
      size 0.075 0.075 0.075
    }
  }
  physics Physics {
    density -1
    mass 0.2
  }
  controller "rus_controller"
  synchronization FALSE
}
Solid {
  translation -0.19 0.0176 -0.26
  rotation 0 1 0 3.14159
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.3 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(4)"
  boundingObject USE parede
}
Solid {
  translation 0.03 0.0176 -0.26
  rotation 0 1 0 3.14159
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.3 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(5)"
  boundingObject USE parede
}
Solid {
  translation 0.18 0.0176 -0.11
  rotation 0 -1 0 -1.5707953071795862
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.3 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(12)"
  boundingObject USE parede
}
Solid {
  translation 0.0399986 0.0176 0.0300001
  rotation 0 1 0 -5.307179586466759e-06
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.3 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(13)"
  boundingObject USE parede
}
Solid {
  translation 0.37 0.0176 -0.32
  rotation 0 1 0 -1.5707953071795862
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.3 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(7)"
  boundingObject USE parede
}
Solid {
  translation 0.37 0.0176 -0.16
  rotation 0 1 0 -1.5707953071795862
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.3 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(17)"
  boundingObject USE parede
}
Solid {
  translation 0.22 0.0176 -0.47
  rotation 0 1 0 3.14159
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.3 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(8)"
  boundingObject USE parede
}
Solid {
  translation -0.06 0.0176 -0.47
  rotation 0 1 0 3.14159
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.3 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(9)"
  boundingObject USE parede
}
Solid {
  translation -0.05 0.0176 -0.11
  rotation 0 1 0 3.14159
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.3 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(16)"
  boundingObject USE parede
}
Solid {
  translation 7.13734e-08 0.0176 -0.62
  rotation 0 -1 0 -1.5707953071795862
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.3 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(14)"
  boundingObject USE parede
}
Solid {
  translation -0.35 0.0176 -0.47
  rotation 1.32679e-06 4.7949e-09 -1 3.14159
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.3 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(10)"
  boundingObject USE parede
}
Solid {
  translation 0.37 0.0176 0.06
  rotation 0 -1 0 -1.5707953071795862
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.3 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(1)"
  boundingObject USE parede
}
Solid {
  translation 0.0500001 0.0176 0.21
  rotation 0 1 0 -5.307179586466759e-06
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.3 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(2)"
  boundingObject USE parede
}
Solid {
  translation -0.19 0.0176 0.19
  rotation 0 1 0 -5.307179586466759e-06
  children [
    Transform {
      children [
        Solid {
          translation 0 0 0.02
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.3 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(6)"
  boundingObject USE parede
}
Solid {
  translation -0.34 0.0176 0.07
  rotation 0 -1 0 -1.5707953071795862
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.3 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(3)"
  boundingObject USE parede
}
Solid {
  translation -0.34 0.0176 -0.12
  rotation 0 -1 0 -1.5707953071795862
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.3 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(11)"
  boundingObject USE parede
}
Solid {
  translation 0.33 0.0176 0.21
  rotation 0 1 0 3.14159
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.3 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(21)"
  boundingObject USE parede
}
